docs/source/upgrade_to_newest_version.md

```diff
@@ -0,0 +1,186 @@
+
+After the `Application` capability has been updated to `V2_0`, you must use
+the v2.0 lifecycle procedures to package, install, approve, and commit new
+chaincodes on the channel. As a result, make sure to be prepared for the new
+lifecycle before updating the capability.
+
+The new lifecycle defaults to using the endorsement policy configured in the
+channel config (e.g., a `MAJORITY` of orgs). Therefore this endorsement policy
+should be added to the channel configuration when enabling capabilities on the
+channel.
+
+For information about how to edit the relevant channel configurations to enable
+the new lifecycle, check out [Enabling the new chaincode lifecycle](./enable_cc_lifecycle.html).
+
+It is also a best practice to add an endorsement policy for each organization,
+which will also define how changes will be made at the organization level.
+
+## Chaincode shim changes
+
+Note that these changes only apply to chaincode written in Go.
+
+If you vendor the shim in v1.4 chaincode before making upgrades to the peers and
+channels, no action is needed for the chaincode. This is the recommended approach.
+
+If you did not vendor the shim in your v1.4 chaincode, the old v1.4 chaincode
+images will still technically work after upgrade, but you are in a risky state.
+If the chaincode image gets deleted from your environment for whatever reason,
+the next invoke on v2.0 peer will try to rebuild the chaincode image and you'll
+get an error that the shim cannot be found.
+
+At this point, you have two options:
+
+1. If the entire channel is ready to upgrade chaincode, you can upgrade the
+   chaincode on all peers and on the channel (using either the old or new
+   lifecycle depending on the `Application` capability level you have enabled).
+   The best practice at this point would be to vendor the new Go chaincode shim
+   using modules.
+
+2. If the entire channel is not yet ready to upgrade the chaincode, you can use
+   peer environment variables to specify the v1.4 chaincode environment `ccenv`
+   be used to rebuild the chaincode images. This v1.4 `ccenv` should still work
+   with a v2.0 peer.
+
+## Chaincode logger (Go chaincode only)
+
+Support for user chaincodes to utilize the chaincode shim's logger via `NewLogger()`.
+Chaincodes that used the shim's `NewLogger()` must now shift to their own
+preferred logging mechanism.
+
+For more information, check out [Logging control](./logging-control.html#chaincode).
+
+## Peer databases upgrade
+
+The state databases of all peers must be rebuilt as part of the upgrade to v2.0.
+To trigger the rebuild, the database must be dropped before the peer is started.
+To do this, you must first export the `FABRIC_CFG_PATH` to point to the `core.yaml`
+file for the peer. Then issue:
+
+```
+peer node upgrade-dbs
+```
+
+Note that this process can take several hours, depending on the size of your
+database. Once that has completed, issue the following command to start the node:
+
```

Comment:
Is there a way to monitor this process and be assured it is not hung?

Reply:
Good question. @denyeart can you comment?

Reply:
@joealewine yes, in the peer log you will see a message like this for every 1000th block, precisely so that you know the peer is progressing:

[lockbasedtxmgr] CommitLostBlock -> INFO 041 Recommitting block [1000] to state database

Reply:
Thanks @denyeart!